packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts

```diff
@@ -1074,6 +1147,35 @@ export class Service extends cdk.Resource implements iam.IGrantable {
       throw new Error('configurationValues cannot be provided if the ConfigurationSource is Repository');
     }
 
+    if (this.props.healthCheckConfiguration?.path !== undefined) {
```

Comment:
The implementation is correct.
I'd create a separate function to handle the logic to keep the constructor cleaner, something like:
```
private validateHealthCheckConfiguration(healthCheckConfiguration?: HealthCheckConfiguration) {
    // Add guard to avoid using ? on every successive control
    if (!healthCheckConfiguration) return; 
    // Validate parameters...
}
```